crates/polars-core/src/chunked_array/ops/apply.rs

```diff
@@ -115,6 +115,55 @@ where
 
         ChunkedArray::try_from_chunk_iter(self.name(), iter)
     }
+
+    pub fn apply_to_buffer_generic<'a, F>(&'a self, mut f: F) -> StringChunked
+    where
+        F: FnMut(T::Physical<'a>, &mut String),
+    {
+        let mut buf = String::new();
+        let chunks = self
+            .downcast_iter()
+            .map(|arr| {
+                let mut mutarr = MutablePlString::with_capacity(arr.len());
+                arr.iter().for_each(|opt| match opt {
+                    None => mutarr.push_null(),
+                    Some(v) => {
+                        buf.clear();
+                        f(v, &mut buf);
+                        mutarr.push_value(&buf)
+                    },
+                });
+                mutarr.freeze()
+            })
+            .collect::<Vec<_>>();
+        ChunkedArray::from_chunk_iter(self.name(), chunks)
+    }
+
+    pub fn try_apply_to_buffer_generic<'a, F, E>(&'a self, mut f: F) -> Result<StringChunked, E>
+    where
+        F: FnMut(T::Physical<'a>, &mut String) -> Result<(), E>,
+    {
+        let mut buf = String::new();
+        let chunks = self
+            .downcast_iter()
+            .map(|arr| {
+                let mut mutarr = MutablePlString::with_capacity(arr.len());
+                arr.iter().try_for_each(|opt| {
```

Comment:
Can we make this just a loop? It's a bit less cluttered with the fallible return types.